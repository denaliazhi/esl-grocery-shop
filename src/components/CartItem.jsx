/**
 * This component renders the details for an item in cart view.
 */
import NumberPicker from "./NumberPicker";
import RemoveItem from "./RemoveItem";
import { capitalize, getImagePath } from "../utils.js";

export default function CartItem({ item, handleClick }) {
  let totalCost = item.count * item.unitPrice;
  return (
    <>
      <div
        className="product cart-item"
        id={item.id}
        data-section={item.section}
      >
        <img src={getImagePath(item.name)} alt="" width="50" />
        <div>
          <p>{capitalize(item.name)},</p>
          <p>
            ${item.unitPrice.toFixed(2)} / {item.displayUnit}
          </p>
          <div className="edit-cart">
            <NumberPicker value={item.count} handleClick={handleClick} />
            <RemoveItem value={item.count} handleClick={handleClick} />
          </div>
        </div>
        <p>${totalCost.toFixed(2)}</p>
      </div>
    </>
  );
}
