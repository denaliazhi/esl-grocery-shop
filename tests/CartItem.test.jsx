import { vi, describe, test, expect } from "vitest";
import { render } from "@testing-library/react";

import CartItem from "../src/components/CartItem";

vi.mock("../src/components/NumberPicker", () => ({
  default: vi.fn(({ value, handleClick }) => <button>beep</button>),
}));

vi.mock("../src/components/RemoveItem", () => ({
  default: vi.fn(({ value, handleClick }) => <button>boop</button>),
}));

describe("Cart item", () => {
  test("renders expected content", () => {
    const item = {
      id: "x123",
      name: "apple",
      unitPrice: 1.23,
      displayUnit: "each",
      section: "produce",
      count: 1,
      totalCost: 1.23,
    };
    const { container } = render(
      <CartItem item={item} handleClick={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });
});
