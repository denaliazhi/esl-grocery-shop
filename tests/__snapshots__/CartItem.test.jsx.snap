// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart item > renders expected content 1`] = `
<div>
  <div
    class="product cart-item"
    data-section="produce"
    id="x123"
  >
    <img
      alt="apple"
      src="products/apple.png"
      width="50"
    />
    <div>
      <p>
        Apple
        ,
      </p>
      <p>
        $
        1.23
         / 
      </p>
      <div
        class="edit-cart"
      >
        <button>
          beep
        </button>
        <button>
          boop
        </button>
      </div>
    </div>
    <p>
      $
      1.23
    </p>
  </div>
</div>
`;
